docs/ReferenceGuides/UnderscoredAttributes.md

```diff
@@ -0,0 +1,131 @@
+# Underscored Attributes Reference
+
+**WARNING:** This information is provided primarily for compiler and standard
+library developers. Usage of these attributes outside of the Swift monorepo
+is strongly discouraged.
+
+The Swift reference has a chapter discussing [stable attributes][Attributes].
+This document is intended to serve as a counter-part describing underscored
+attributes, whose semantics are subject to change and most likely need to
+go through the Swift evolution process before being stabilized.
+
+The attributes are organized in alphabetical order.
+
+## `@_alignment`
+
+## `@_borrowed`
+
+## `@_cdecl`
+
+Similar to `@_silgen_name` but uses the C calling convention.
+This attribute doesn't have very well-defined semantics.
+
+For potential ideas on stabilization, see
+[Formalizing `@cdecl`](https://forums.swift.org/t/formalizing-cdecl/40677).
+
```

Comment:
## `@_disfavoredOverload`

Marks a declaration that the type checker should try to avoid using. When the
expression type checker is considering overloads, it will always prefer a solution
with fewer `@_disfavoredOverload` declarations over one with more of them.

This was designed to help with `ExpressibleBy*Literal` types. The type checker
strongly prefers to give literals their default type (e.g. `Int` for `ExpressibleByIntegerLiteral`,
`String` for `ExpressibleByStringLiteral`, etc.), so if you want your API to prefer some other
type but accept the default one too, you may need to mark the declaration taking the default
literal type with this attribute:

    extension LocalizedStringKey: ExpressibleByStringLiteral { ... }
    
    extension Text {
        // We want `Text("foo")` to use this initializer:
        init(_ key: LocalizedStringKey) { ... }
    
        // But without @_disfavoredOverload, it would use this one instead:
        @_disfavoredOverload init<S: StringProtocol>(_ str: S) { ... }
    }

`@_disfavoredOverload` is a stopgap design and we want to eventually fix
the type checker behavior and then remove the attribute or make it a no-op.
Use it only to work around obviously broken overload resolution behavior,
not to arbitrarily manipulate overload priority, and be prepared for it to break
without warning.